sherpa/plot/pylab_backend.py

```diff
@@ -211,9 +205,9 @@ def histo(xlo, xhi, y, yerr=None, title=None, xlabel=None, ylabel=None,
           overplot=False, clearwindow=True,
           xerrorbars=False,
           yerrorbars=False,
-          ecolor=_errorbar_defaults['ecolor'],
-          capsize=_errorbar_defaults['capsize'],
-          barsabove=_errorbar_defaults['barsabove'],
+          ecolor=None,
+          capsize=None,
+          barsabove=False,
```

Comment:
Is the idea here being that the defaults for `plt.errorbar` are unlikely to change so we can just hard-code them?

If you git-blame the `_errorbar_defaults` setting you can see it was written a long time ago, in a galaxy far away, when matplotlib was significantly younger, so it may have made sense to make the values dependent on the backend.

This is also the only "plot object" that has this sort of functionality, so I don't think oit's a bad change, just checking what the thinking was.

Reply:
It's a step towards allowing full functionality without matplotlib present. `_errorbar_defaults` looks at the signature of `plt.errorbar`, so `plt` must be around for that to work. I want to move to (but this is just one of the tiny first steps that way) a set of defaults that don't require importing a plotting library first. For the reasons you listed above, hardcoding it here is fine in my opinion, and `ecolor=None` actually means "default" (as defined in the matplotlib style file) in matplotlib language, so this is also a way to allow users to influence our plots with matplotlic style files if they want to (again, that's just a thought that that's a good direction to go in, I'm not claiming that these three lines of code are a full solution)

Reply:
Getting away from the actual change here,but I'm confused. This module requires matplotlib to exist, so I don't see how it helps in your overall goal of decoupling from the specific backend, but maybe that needs to come from a follow-on PR.

Reply:
Mm, @DougBurke is right. What I said above makes no sense, matplotlib is (and always will be) required for this backend. I'm sorry - I wrote this code many months ago and must have mixed up up this change with what I had thought about for some other change. However, we agree that this is not a bad change, even if I can't recall exactly why I did it initially, so I'll leave it. At the very least (and that might have well been my motivation) it simplifies the module by removing and additional look-up that one has to track mentally.